lib/ufe/StagesSubject.cpp

```diff
@@ -223,5 +254,39 @@ void StagesSubject::onStageSet(const UsdMayaProxyStageSetNotice& notice)
 	afterOpen();
 }
 
+AttributeChangedNotificationGuard::AttributeChangedNotificationGuard()
+{
+    if (inAttributeChangedNotificationGuard()) {
+        TF_CODING_ERROR("Attribute changed notification guard cannot be nested.");
+    }
+
+    if (attributeChangedNotificationGuardCount == 0 &&
+        !pendingAttributeChangedNotifications.empty()) {
+        TF_CODING_ERROR("Stale pending attribute changed notifications.");
+    }
+
+    ++attributeChangedNotificationGuardCount;
+
+}
+
+AttributeChangedNotificationGuard::~AttributeChangedNotificationGuard()
+{
+    --attributeChangedNotificationGuardCount;
+
+    if (attributeChangedNotificationGuardCount < 0) {
+        TF_CODING_ERROR("Corrupt attribute changed notification guard.");
+    }
+
+    if (attributeChangedNotificationGuardCount > 0 ) {
+        return;
+    }
+
+    for (const auto& notificationInfo : pendingAttributeChangedNotifications) {
```

Comment:
On guard expiration send out collected notifications.  With current use of the guard, this will be a loop of a single element, but we are ready for any more general use.